### PR TITLE
chore: fix a few typos

### PR DESCRIPTION
**Pull request by Lioness100**
This PR fixes a few typos throughout the repository. I didn't test or lint because `mongodb-memory-server` wasn't building for me locally and I needed to get to sleep haha.

### CODE REVIEW

**Review by hasezoey — CHANGES_REQUESTED** [review]
Thanks for finding and correcting these typos, but i have to request some changes:
- when changing strings in tests or code, make sure the characters are properly escaped or change the string type
- if possible, maybe split these changes out into multiple commits (like docs typo fixes and code fixes)

**Comment by hasezoey** [comment]
> I didn't test or lint because mongodb-memory-server wasn't building for me locally and I needed to get to sleep haha.

could you maybe post the errors and your system information? (because i am also a maintainer of that package)
- mongodb-memory-server version (and which package is used)
- your system (mac/linux (with distro)/windows)
- arch (x86_64/arm64)

**Comment by Lioness100** [comment]
Hi! 

- Just to make sure I didn't miss one, the only string I forgot to escape is in `src/internal/errors.ts`, right?

- Do you want 2 separate PRs? Or just two different commits for more organized viewing

- I didn't get any errors, it was just stuck on something like "Building mongodb-memory-server" for ~7 minutes until I gave up. System: Windows 11 x64

**Comment by hasezoey** [comment]
> Just to make sure I didn't miss one, the only string I forgot to escape is in src/internal/errors.ts, right?

i think so

> Do you want 2 separate PRs? Or just two different commits for more organized viewing

just different commits if possible, it makes it cleaner (in my opinion)

> I didn't get any errors, it was just stuck on something like "Building mongodb-memory-server" for ~7 minutes until I gave up. System: Windows 11 x64

weird, if you can reproduce it, could you try again with `MONGOMS_DEBUG=1 npm install --foreground-scripts` (or if yarn `MONGOMS_DEBUG=1 npm install`) and post the full output? (or if not in a install, just add `MONGOMS_DEBUG=1` infront)

Note: i dont know if this is the proper way to set environment variables on non UNIX systems

---

forgot to add in my last comment:
>  I didn't test or lint because mongodb-memory-server wasn't building for me locally and I needed to get to sleep haha.

MMS is only required for tests, it does not affect linting (though if it was the postinstall failing, you can try to reinstall everything with `MONGOMS_DISABLE_POSTINSTALL=1`

also, MMS is not *strictly* required for testing, as long as you have a mongod server already running that can be used for tests (though you need to set the config properly, which can be more work than its worth for a typo fix)

**Comment by hasezoey** [comment]
>  I've been trying for a while and totally entrapped myself in a web of git messups lol sob I'm not great at git :(

thats fine then, it was a optional request anyway

---

only things needs then would be to fix the current syntax problems, then making sure the tests pass and any lint fixes if there are any

**Comment by Lioness100** [comment]
Lint and tests pass locally :+1:

File: CHANGELOG.md [hunk 1]
```diff
@@ -1638,15 +1638,15 @@ Everything Below here is manually made:
 - Add Error Codes [Errors & Warnings Details](guides/error-warnings-details.md)
 - Remove never triggered Error `InvalidPropError`
 - Change "Options-not-for-current-type" Errors into warnings (with actual information on what options are included) [[typegoose#363](https://github.com/typegoose/typegoose/issues/363)]
-- [IC] Assing `schemaOptions` in `src/internal/schema.ts` to an black object [[typegoose#357](https://github.com/typegoose/typegoose/issues/357)]
+- [IC] Adding `schemaOptions` in `src/internal/schema.ts` to an black object [[typegoose#357](https://github.com/typegoose/typegoose/issues/357)]
```
Comment (hasezoey) [inline CHANGELOG.md:1641]:
```suggestion
- [IC] Assigning `schemaOptions` in `src/internal/schema.ts` to an black object [[typegoose#357](https://github.com/typegoose/typegoose/issues/357)]
```

**Review by hasezoey — APPROVED** [review]
just 1 minor thing left (will self-apply):
- reverting wording for one changelog entry (just wanna keep the same meaning)

File: CHANGELOG.md (hunk 1 above)
Comment (Lioness100) [reply CHANGELOG.md]: Ah, that makes sense, I wasn't sure about that one

**Pull request merged**